diff -r a72d9ca94ce2 org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/custom/CTabFolder.d
--- a/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/custom/CTabFolder.d	Sat Dec 24 17:16:18 2011 +0900
+++ b/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/custom/CTabFolder.d	Sat Dec 24 18:02:52 2011 +0900
@@ -119,7 +119,7 @@
      *
      * @deprecated drop shadow border is no longer drawn in 3.0
      */
-    mixin(gshared!(`public static RGB borderInsideRGB;`));
+    __gshared public static RGB borderInsideRGB;
     /**
      * Color of middle line of drop shadow border.
      *
@@ -128,7 +128,7 @@
      *
      * @deprecated drop shadow border is no longer drawn in 3.0
      */
-    mixin(gshared!(`public static RGB borderMiddleRGB;`));
+    __gshared public static RGB borderMiddleRGB;
     /**
      * Color of outermost line of drop shadow border.
      *
@@ -137,7 +137,7 @@
      *
      * @deprecated drop shadow border is no longer drawn in 3.0
      */
-    mixin(gshared!(`public static RGB borderOutsideRGB;`));
+    __gshared public static RGB borderOutsideRGB;
 
     /* sizing, positioning */
     int xClient, yClient;
@@ -278,19 +278,19 @@
     static const int NORMAL = 1;
     static const int HOT = 2;
     static const int SELECTED = 3;
-    mixin(gshared!(`static const RGB CLOSE_FILL;`));
+    __gshared static const RGB CLOSE_FILL;
 
     static const int CHEVRON_CHILD_ID = 0;
     static const int MINIMIZE_CHILD_ID = 1;
     static const int MAXIMIZE_CHILD_ID = 2;
     static const int EXTRA_CHILD_ID_COUNT = 3;
 
-mixin(sharedStaticThis!(`{
+shared static this(){
     borderInsideRGB  = new RGB (132, 130, 132);
     borderMiddleRGB  = new RGB (143, 141, 138);
     borderOutsideRGB = new RGB (171, 168, 165);
     CLOSE_FILL = new RGB(252, 160, 160);
-}`));
+}
 
 /**
  * Constructs a new instance of this class given its parent
diff -r a72d9ca94ce2 org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/custom/StyledText.d
--- a/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/custom/StyledText.d	Sat Dec 24 17:16:18 2011 +0900
+++ b/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/custom/StyledText.d	Sat Dec 24 18:02:52 2011 +0900
@@ -239,13 +239,13 @@
     int indent;
     int lineSpacing;
 
-    mixin(gshared!(`const static bool IS_CARBON, IS_GTK, IS_MOTIF;`));
-    mixin(sharedStaticThis!(`{
+    __gshared const static bool IS_CARBON, IS_GTK, IS_MOTIF;
+    shared static this(){
         String platform = SWT.getPlatform();
         IS_CARBON = ("carbon" == platform);
         IS_GTK    = ("gtk"    == platform);
         IS_MOTIF  = ("motif"  == platform);
-    }`));
+    }
 
     /**
      * The Printing class : printing of a range of text.
diff -r a72d9ca94ce2 org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/custom/ViewForm.d
--- a/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/custom/ViewForm.d	Sat Dec 24 17:16:18 2011 +0900
+++ b/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/custom/ViewForm.d	Sat Dec 24 18:02:52 2011 +0900
@@ -92,7 +92,7 @@
      *
      * @deprecated
      */
-    mixin(gshared!(`public static RGB borderInsideRGB;`));
+    __gshared public static RGB borderInsideRGB;
     /**
      * Color of middle line of drop shadow border.
      *
@@ -101,7 +101,7 @@
      *
      * @deprecated
      */
-    mixin(gshared!(`public static RGB borderMiddleRGB;`));
+    __gshared public static RGB borderMiddleRGB;
     /**
      * Color of outermost line of drop shadow border.
      *
@@ -110,7 +110,7 @@
      *
      * @deprecated
      */
-    mixin(gshared!(`public static RGB borderOutsideRGB;`));
+    __gshared public static RGB borderOutsideRGB;
 
     // SWT widgets
     Control topLeft;
@@ -137,11 +137,11 @@
     static const int SELECTION_BACKGROUND = SWT.COLOR_LIST_BACKGROUND;
 
 
-    mixin(sharedStaticThis!(`{
+    shared static this(){
         borderInsideRGB  = new RGB (132, 130, 132);
         borderMiddleRGB  = new RGB (143, 141, 138);
         borderOutsideRGB = new RGB (171, 168, 165);
-    }`));
+    }
 /**
  * Constructs a new instance of this class given its parent
  * and a style value describing its behavior and appearance.
diff -r a72d9ca94ce2 org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/dnd/DragSource.d
--- a/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/dnd/DragSource.d	Sat Dec 24 17:16:18 2011 +0900
+++ b/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/dnd/DragSource.d	Sat Dec 24 18:02:52 2011 +0900
@@ -146,11 +146,11 @@
     int dataEffect = DND.DROP_NONE;
 
     static const String DEFAULT_DRAG_SOURCE_EFFECT = "DEFAULT_DRAG_SOURCE_EFFECT"; //$NON-NLS-1$
-    mixin(gshared!(`static const int CFSTR_PERFORMEDDROPEFFECT;`));
+    __gshared static const int CFSTR_PERFORMEDDROPEFFECT;
     static const TCHAR[] WindowClass = "#32770\0";
-    mixin(sharedStaticThis!(`{
+    shared static this(){
         CFSTR_PERFORMEDDROPEFFECT  = Transfer.registerType("Performed DropEffect");     //$NON-NLS-1$
-    }`));
+    }
 /**
  * Creates a new <code>DragSource</code> to handle dragging from the specified <code>Control</code>.
  * Creating an instance of a DragSource may cause system resources to be allocated depending on the platform.
diff -r a72d9ca94ce2 org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/dnd/HTMLTransfer.d
--- a/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/dnd/HTMLTransfer.d	Sat Dec 24 17:16:18 2011 +0900
+++ b/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/dnd/HTMLTransfer.d	Sat Dec 24 18:02:52 2011 +0900
@@ -40,7 +40,7 @@
 
     static HTMLTransfer _instance;
     static const String HTML_FORMAT = "HTML Format"; //$NON-NLS-1$
-    mixin(gshared!(`static const int HTML_FORMATID;`));
+    __gshared static const int HTML_FORMATID;
     static const String NUMBER = "00000000"; //$NON-NLS-1$
     static const String HEADER = "Version:0.9\r\nStartHTML:"~NUMBER~"\r\nEndHTML:"~NUMBER~"\r\nStartFragment:"~NUMBER~"\r\nEndFragment:"~NUMBER~"\r\n";
     static const String PREFIX = "<html><body><!--StartFragment-->"; //$NON-NLS-1$
@@ -48,9 +48,9 @@
     static const String StartFragment = "StartFragment:"; //$NON-NLS-1$
     static const String EndFragment = "EndFragment:"; //$NON-NLS-1$
 
-mixin(sharedStaticThis!(`{
+shared static this(){
     HTML_FORMATID = registerType(HTML_FORMAT);
-}`));
+}
 
 private this() {}
 
diff -r a72d9ca94ce2 org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/dnd/ImageTransfer.d
--- a/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/dnd/ImageTransfer.d	Sat Dec 24 17:16:18 2011 +0900
+++ b/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/dnd/ImageTransfer.d	Sat Dec 24 18:02:52 2011 +0900
@@ -42,13 +42,13 @@
  */
 public class ImageTransfer : ByteArrayTransfer {
 
-    mixin(gshared!(`private static ImageTransfer _instance;`));
+    __gshared private static ImageTransfer _instance;
     private static const String CF_DIB = "CF_DIB"; //$NON-NLS-1$
     private static const int CF_DIBID = COM.CF_DIB;
 
-mixin(sharedStaticThis!(`{
+shared static this(){
      _instance = new ImageTransfer();
-}`));
+}
 
 private this() {
 }
diff -r a72d9ca94ce2 org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/dnd/RTFTransfer.d
--- a/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/dnd/RTFTransfer.d	Sat Dec 24 17:16:18 2011 +0900
+++ b/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/dnd/RTFTransfer.d	Sat Dec 24 18:02:52 2011 +0900
@@ -41,11 +41,11 @@
 
     private static RTFTransfer _instance;
     private static const String CF_RTF = "Rich Text Format"; //$NON-NLS-1$
-    mixin(gshared!(`private static const int CF_RTFID;`));
+    __gshared private static const int CF_RTFID;
 
-mixin(sharedStaticThis!(`{
+shared static this(){
     CF_RTFID = registerType(CF_RTF);
-}`));
+}
 
 private this() {}
 
diff -r a72d9ca94ce2 org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/dnd/URLTransfer.d
--- a/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/dnd/URLTransfer.d	Sat Dec 24 17:16:18 2011 +0900
+++ b/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/dnd/URLTransfer.d	Sat Dec 24 18:02:52 2011 +0900
@@ -40,11 +40,11 @@
 
     static URLTransfer _instance;
     static const String CFSTR_INETURL = "UniformResourceLocator"; //$NON-NLS-1$
-    mixin(gshared!(`static const int CFSTR_INETURLID;`));
+    __gshared static const int CFSTR_INETURLID;
 
-mixin(sharedStaticThis!(`{
+shared static this(){
     CFSTR_INETURLID = registerType(CFSTR_INETURL);
-}`));
+}
 
 private this() {}
 
diff -r a72d9ca94ce2 org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/graphics/Device.d
--- a/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/graphics/Device.d	Sat Dec 24 17:16:18 2011 +0900
+++ b/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/graphics/Device.d	Sat Dec 24 18:02:52 2011 +0900
@@ -103,8 +103,8 @@
     *
     * This code will be removed in the future.
     */
-    mixin(gshared!(`protected static Device CurrentDevice;`));
-    mixin(gshared!(`protected static Runnable DeviceFinder;`));
+    __gshared protected static Device CurrentDevice;
+    __gshared protected static Runnable DeviceFinder;
 
 
 /*
diff -r a72d9ca94ce2 org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/graphics/ImageData.d
--- a/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/graphics/ImageData.d	Sat Dec 24 17:16:18 2011 +0900
+++ b/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/graphics/ImageData.d	Sat Dec 24 18:02:52 2011 +0900
@@ -222,11 +222,11 @@
     /**
      * Arbitrary channel width data to 8-bit conversion table.
      */
-    mixin(gshared!(`private static byte[][] ANY_TO_EIGHT;`));
-    mixin(gshared!(`private static byte[] ONE_TO_ONE_MAPPING;`));
+    __gshared private static byte[][] ANY_TO_EIGHT;
+    __gshared private static byte[] ONE_TO_ONE_MAPPING;
 
-    mixin(gshared!(`private static bool static_this_completed = false;`));
-    mixin(sharedStatic_This!(`{
+    __gshared private static bool static_this_completed = false;
+    shared private static void static_this(){
         if( static_this_completed ) return;
         synchronized {
             if( static_this_completed ) return;
@@ -241,7 +241,7 @@
             ONE_TO_ONE_MAPPING = ANY_TO_EIGHT[8];
             static_this_completed = true;
         }
-    }`));
+    }
 
     /**
      * Scaled 8x8 Bayer dither matrix.
diff -r a72d9ca94ce2 org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/graphics/TextLayout.d
--- a/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/graphics/TextLayout.d	Sat Dec 24 17:16:18 2011 +0900
+++ b/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/graphics/TextLayout.d	Sat Dec 24 18:02:52 2011 +0900
@@ -125,7 +125,7 @@
     static byte[16] IID_IMLangFontLink2;
     static bool static_this_completed = false;
 "));
-    mixin(sharedStatic_This!(`{
+    shared private static void static_this(){
         // in case of allready initialized, we can check and leave without lock
         if( static_this_completed ){
             return;
@@ -137,7 +137,7 @@
                 static_this_completed = true;
             }
         }
-    }`));
+    }
 
     /* IME has a copy of these constants */
     static const int UNDERLINE_IME_DOT = 1 << 16;
diff -r a72d9ca94ce2 org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/internal/gdip/Gdip.d
--- a/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/internal/gdip/Gdip.d	Sat Dec 24 17:16:18 2011 +0900
+++ b/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/internal/gdip/Gdip.d	Sat Dec 24 18:02:52 2011 +0900
@@ -40,11 +40,11 @@
 
 public class Gdip : Platform
 {
-    mixin(sharedStaticThis!(`{
+    shared static this(){
         if (!OS.IsWinCE && OS.WIN32_VERSION >= OS.VERSION (5, 1)) {
             loadLib_Gdip();
         }
-    }`));
+    }
     /**************************************************************************
 
     **************************************************************************/
diff -r a72d9ca94ce2 org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/internal/gdip/native.d
--- a/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/internal/gdip/native.d	Sat Dec 24 17:16:18 2011 +0900
+++ b/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/internal/gdip/native.d	Sat Dec 24 18:02:52 2011 +0900
@@ -1198,8 +1198,8 @@
 Status function(Handle format, in int hotkeyPrefix) GdipSetStringFormatHotkeyPrefix;
 Status function(Handle format, in float firstTabOffset, in int count, in float* tabStops) GdipSetStringFormatTabStops;
 
-mixin(gshared!(`Symbol[] symbols;`));
-mixin(sharedStaticThis!(`{
+__gshared Symbol[] symbols;
+shared static this(){
     symbols = [
         Symbol( "GdiplusStartup", cast(void**)& GdiplusStartup ),
         Symbol( "GdiplusShutdown", cast(void**)& GdiplusShutdown ),
@@ -1671,7 +1671,7 @@
         Symbol( "GdipSetStringFormatHotkeyPrefix", cast(void**)& GdipSetStringFormatHotkeyPrefix ),
         Symbol( "GdipSetStringFormatTabStops", cast(void**)& GdipSetStringFormatTabStops )
     ];
-}`));
+}
 
 
 void loadLib_Gdip(){
diff -r a72d9ca94ce2 org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/internal/image/JPEGFileFormat.d
--- a/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/internal/image/JPEGFileFormat.d	Sat Dec 24 17:16:18 2011 +0900
+++ b/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/internal/image/JPEGFileFormat.d	Sat Dec 24 18:02:52 2011 +0900
@@ -126,7 +126,7 @@
     public static const int ID_Y        = 1 - 1;
     public static const int ID_CB   = 2 - 1;
     public static const int ID_CR   = 3 - 1;
-    mixin(gshared!(`public static /*const*/ RGB[] RGB16;`));
+    __gshared public static /*const*/ RGB[] RGB16;
     public static const int[] ExtendTest = [
         0, 1, 2, 4, 8, 16, 32, 64, 128, 256, 512, 1024, 2048,
         4096, 8192, 16384, 32768, 65536, 131072, 262144
@@ -146,11 +146,11 @@
         53, 60, 61, 54, 47, 55, 62, 63
     ];
 
-    mixin(gshared!(`public static int[] CrRTable, CbBTable, CrGTable, CbGTable;`));
-    mixin(gshared!(`public static int[] RYTable, GYTable, BYTable,
-        RCbTable, GCbTable, BCbTable, RCrTable, GCrTable, BCrTable, NBitsTable;`));
+    __gshared public static int[] CrRTable, CbBTable, CrGTable, CbGTable;
+    __gshared public static int[] RYTable, GYTable, BYTable,
+        RCbTable, GCbTable, BCbTable, RCrTable, GCrTable, BCrTable, NBitsTable;
     //public static void static_this() {
-    mixin(sharedStaticThis!(`{
+    shared static this(){
 
         RGB16 = [
             new RGB(0,0,0),
@@ -229,7 +229,7 @@
             nBitsTable[i] = nBits;
         }
         NBitsTable = nBitsTable;
-    }`));
+    }
 void compress(ImageData image, byte[] dataYComp, byte[] dataCbComp, byte[] dataCrComp) {
     int srcWidth = image.width;
     int srcHeight = image.height;
diff -r a72d9ca94ce2 org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/internal/image/PngChunk.d
--- a/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/internal/image/PngChunk.d	Sat Dec 24 17:16:18 2011 +0900
+++ b/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/internal/image/PngChunk.d	Sat Dec 24 18:02:52 2011 +0900
@@ -50,13 +50,13 @@
     static const byte[] TYPE_IEND = cast(byte[])"IEND";//{(byte) 'I', (byte) 'E', (byte) 'N', (byte) 'D'};
     static const byte[] TYPE_tRNS = cast(byte[])"tRNS";//{(byte) 't', (byte) 'R', (byte) 'N', (byte) 'S'};
 
-    mixin(gshared!(`private static int[] _CRC_TABLE = null;`));
+    __gshared private static int[] _CRC_TABLE = null;
     static int[] CRC_TABLE() {
         if (!_CRC_TABLE) static_this();
         return _CRC_TABLE;
     }
     //public static void static_this() {
-    mixin(sharedStatic_This!(`{
+    shared private static void static_this(){
         _CRC_TABLE = new int[256];
         for (int i = 0; i < 256; i++) {
             _CRC_TABLE[i] = i;
@@ -68,7 +68,7 @@
                 }
             }
         }
-    }`));
+    }
 
     int length;
 
diff -r a72d9ca94ce2 org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/internal/image/PngDeflater.d
--- a/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/internal/image/PngDeflater.d	Sat Dec 24 17:16:18 2011 +0900
+++ b/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/internal/image/PngDeflater.d	Sat Dec 24 18:02:52 2011 +0900
@@ -121,10 +121,10 @@
 
 }
 
-mixin(gshared!(`static TryConst!(Code[]) lengthCodes;`));
-mixin(gshared!(`static TryConst!(Code[]) distanceCodes;`));
+__gshared static TryConst!(Code[]) lengthCodes;
+__gshared static TryConst!(Code[]) distanceCodes;
 
-mixin(sharedStaticThis!(`{
+shared static this(){
     lengthCodes = [
         new Code(257, 0, 3, 3),
         new Code(258, 0, 4, 4),
@@ -187,7 +187,7 @@
         new Code(27, 12, 12289, 16384),
         new Code(28, 13, 16385, 24576),
         new Code(29, 13, 24577, 32768)];
-}`));
+}
 
 void writeShortLSB(ByteArrayOutputStream baos, int theShort) {
 
diff -r a72d9ca94ce2 org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/internal/win32/OS.d
--- a/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/internal/win32/OS.d	Sat Dec 24 17:16:18 2011 +0900
+++ b/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/internal/win32/OS.d	Sat Dec 24 18:02:52 2011 +0900
@@ -76,8 +76,8 @@
 public class LDWTRESULT {
     public int value;
     // initalize ONE and ZERO in static OS.this();
-    mixin(gshared!(`public static LDWTRESULT ONE;`));
-    mixin(gshared!(`public static LDWTRESULT ZERO;`));
+    __gshared public static LDWTRESULT ONE;
+    __gshared public static LDWTRESULT ZERO;
     public this (int value) { this.value = value; }
 }
 
@@ -150,25 +150,25 @@
     /*
     * SWT Windows flags
     */
-    mixin(gshared!(`public static BOOL IsWin32s;`));
-    mixin(gshared!(`public static BOOL IsWin95;`));
-    mixin(gshared!(`public static BOOL IsWinNT;`));
+    __gshared public static BOOL IsWin32s;
+    __gshared public static BOOL IsWin95;
+    __gshared public static BOOL IsWinNT;
 
     version(WinCE) {
-        mixin(gshared!(`public const static BOOL IsWinCE = true;`));
-        mixin(gshared!(`public static const BOOL IsHPC   = false`)); //todo
+        __gshared public const static BOOL IsWinCE = true;
+        __gshared public static const BOOL IsHPC   = false; //todo
     }
     else {
-        mixin(gshared!(`public const static BOOL IsWinCE = false;`));
-        mixin(gshared!(`public static const BOOL IsHPC   = false;`));
+        __gshared public const static BOOL IsWinCE = false;
+        __gshared public static const BOOL IsHPC   = false;
     }
 
-    mixin(gshared!(`public static const BOOL IsPPC = false;`));
+    __gshared public static const BOOL IsPPC = false;
 
     // PORTING_FIXME, is it Windows WFSP?
-    mixin(gshared!(`public static const BOOL IsSP = false;`));
+    __gshared public static const BOOL IsSP = false;
 
-    mixin(gshared!(`public static const BOOL IsDBLocale;`));
+    __gshared public static const BOOL IsDBLocale;
 
     version(ANSI) {
         public const BOOL IsUnicode = false;
@@ -176,9 +176,9 @@
         public const BOOL IsUnicode = true;
     }
 
-    mixin(gshared!(`public static const int WIN32_MAJOR, WIN32_MINOR, WIN32_VERSION;`));
-    mixin(gshared!(`public static const int COMCTL32_MAJOR, COMCTL32_MINOR, COMCTL32_VERSION;`));
-    mixin(gshared!(`public static const int SHELL32_MAJOR, SHELL32_MINOR, SHELL32_VERSION;`));
+    __gshared public static const int WIN32_MAJOR, WIN32_MINOR, WIN32_VERSION;
+    __gshared public static const int COMCTL32_MAJOR, COMCTL32_MINOR, COMCTL32_VERSION;
+    __gshared public static const int SHELL32_MAJOR, SHELL32_MINOR, SHELL32_VERSION;
 
     public static const char[] NO_MANIFEST = "org.eclipse.swt.internal.win32.OS.NO_MANIFEST";
 
@@ -229,7 +229,7 @@
 //     }
 
     /* Get the Windows version and the flags */
-    mixin(sharedStaticThis!(`{
+    shared static this(){
         static_this_1();
         static_this_2();
         static_this_3();
@@ -376,7 +376,7 @@
         SHELL32_MAJOR = dvi.dwMajorVersion;
         SHELL32_MINOR = dvi.dwMinorVersion;
         SHELL32_VERSION = VERSION (SHELL32_MAJOR, SHELL32_MINOR);
-    }`));
+    }
 
     /**************************************************************************
 
@@ -2675,7 +2675,7 @@
     ) SetWindowTheme;
 
 } // public static
-mixin(gshared!(`static SymbolVersioned2[] Symbols_UxTheme;`));
+__gshared static SymbolVersioned2[] Symbols_UxTheme;
 private static void static_this_1(){
     Symbols_UxTheme = [
         SymbolVersioned2( "IsAppThemed", cast(void**)& IsAppThemed, 5, 1 ),
@@ -2726,8 +2726,8 @@
 } // public static extern(Windows)
 
 
-mixin(gshared!(`static SymbolVersioned2[] Symbols_CoreImm = [
-];`));
+__gshared static SymbolVersioned2[] Symbols_CoreImm = [
+];
 
 // user32.dll vista
 public static extern(Windows){
@@ -2735,7 +2735,7 @@
     BOOL function( HWND hWnd )IsHungAppWindow;
 }
 
-mixin(gshared!(`static SymbolVersioned2[] Symbols_User32;`));
+__gshared static SymbolVersioned2[] Symbols_User32;
 private static void static_this_2(){
     Symbols_User32 = [
         SymbolVersioned2( "SetProcessDPIAware", cast(void**)& SetProcessDPIAware, 6, 0 ),
@@ -2799,7 +2799,7 @@
 
 }
 
-mixin(gshared!(`static SymbolVersioned2[] Symbols_Imm32;`));
+__gshared static SymbolVersioned2[] Symbols_Imm32;
 private static void static_this_3(){
     Symbols_Imm32 = [
         SymbolVersioned2( "ImmAssociateContext", cast(void**)& ImmAssociateContext, 5, 1 ),
@@ -2826,7 +2826,7 @@
 
 version(ANSI){
 }else{
-mixin(gshared!(`static SymbolVersioned2[] Symbols_Kernel32;`));
+__gshared static SymbolVersioned2[] Symbols_Kernel32;
 private static void static_this_4(){
     Symbols_Kernel32 = [
         SymbolVersioned2( "CreateActCtxW", cast(void**)& CreateActCtx, 5, 1 ),
diff -r a72d9ca94ce2 org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/printing/Printer.d
--- a/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/printing/Printer.d	Sat Dec 24 17:16:18 2011 +0900
+++ b/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/printing/Printer.d	Sat Dec 24 18:02:52 2011 +0900
@@ -76,14 +76,14 @@
     /**
      * strings used to access the Windows registry
      */
-    mixin(gshared!(`static StringT profile;`));
-    mixin(gshared!(`static StringT appName;`));
-    mixin(gshared!(`static StringT keyName;`));
-    mixin(sharedStaticThis!(`{
+    __gshared static StringT profile;
+    __gshared static StringT appName;
+    __gshared static StringT keyName;
+    shared static this(){
         profile = StrToTCHARs(0, "PrinterPorts", true); //$NON-NLS-1$
         appName = StrToTCHARs(0, "windows", true); //$NON-NLS-1$
         keyName = StrToTCHARs(0, "device", true); //$NON-NLS-1$
-    }`));
+    }
 
 /**
  * Returns an array of <code>PrinterData</code> objects
diff -r a72d9ca94ce2 org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/Button.d
--- a/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/Button.d	Sat Dec 24 17:16:18 2011 +0900
+++ b/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/Button.d	Sat Dec 24 18:02:52 2011 +0900
@@ -69,14 +69,14 @@
     ImageList imageList;
     bool ignoreMouse, grayed;
     static const int MARGIN = 4;
-    mixin(gshared!(`private static /+const+/ int CHECK_WIDTH, CHECK_HEIGHT;`));
+    __gshared private static /+const+/ int CHECK_WIDTH, CHECK_HEIGHT;
     static const int ICON_WIDTH = 128, ICON_HEIGHT = 128;
     static const bool COMMAND_LINK = false;
-    mixin(gshared!(`private static /+const+/ TWindowProc ButtonProc;`));
+    __gshared private static /+const+/ TWindowProc ButtonProc;
     static const TCHAR[] ButtonClass = "BUTTON\0";
 
-    mixin(gshared!(`private static bool static_this_completed = false;`));
-    mixin(sharedStatic_This!(`{
+    __gshared private static bool static_this_completed = false;
+    shared private static void static_this(){
         if( static_this_completed ){
             return;
         }
@@ -100,7 +100,7 @@
             ButtonProc = lpWndClass.lpfnWndProc;
             static_this_completed = true;
         }
-    }`));
+    }
 
 /**
  * Constructs a new instance of this class given its parent
diff -r a72d9ca94ce2 org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/Combo.d
--- a/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/Combo.d	Sat Dec 24 17:16:18 2011 +0900
+++ b/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/Combo.d	Sat Dec 24 18:02:52 2011 +0900
@@ -114,11 +114,11 @@
     static const int CBID_EDIT = 1001;
     static /*final*/ WNDPROC EditProc, ListProc;
 
-    mixin(gshared!(`static /+const+/ WNDPROC ComboProc;`));
+    __gshared static /+const+/ WNDPROC ComboProc;
     static const TCHAR* ComboClass = "COMBOBOX\0";
 
-    mixin(gshared!(`private static bool static_this_completed = false;`));
-    mixin(sharedStatic_This!(`{
+    __gshared private static bool static_this_completed = false;
+    shared private static void static_this(){
         if( static_this_completed ){
             return;
         }
@@ -131,7 +131,7 @@
             ComboProc = lpWndClass.lpfnWndProc;
             static_this_completed = true;
         }
-    }`));
+    }
 
 
 /**
diff -r a72d9ca94ce2 org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/CoolBar.d
--- a/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/CoolBar.d	Sat Dec 24 17:16:18 2011 +0900
+++ b/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/CoolBar.d	Sat Dec 24 18:02:52 2011 +0900
@@ -66,10 +66,10 @@
     bool locked;
     bool ignoreResize;
     private static /+const+/ WNDPROC ReBarProc;
-    mixin(gshared!(`static const TCHAR* ReBarClass = OS.REBARCLASSNAME.ptr;`));
+    __gshared static const TCHAR* ReBarClass = OS.REBARCLASSNAME.ptr;
 
-    mixin(gshared!(`private static bool static_this_completed = false;`));
-    mixin(sharedStatic_This!(`{
+    __gshared private static bool static_this_completed = false;
+    shared private static void static_this(){
         if( static_this_completed ){
             return;
         }
@@ -86,7 +86,7 @@
             ReBarProc = lpWndClass.lpfnWndProc;
             static_this_completed = true;
         }
-    }`));
+    }
 
     static const int SEPARATOR_WIDTH = 2;
     static const int MAX_WIDTH = 0x7FFF;
diff -r a72d9ca94ce2 org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/DateTime.d
--- a/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/DateTime.d	Sat Dec 24 17:16:18 2011 +0900
+++ b/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/DateTime.d	Sat Dec 24 18:02:52 2011 +0900
@@ -63,13 +63,13 @@
     bool ignoreSelection;
     SYSTEMTIME* lastSystemTime;
     SYSTEMTIME time; // only used in calendar mode
-    mixin(gshared!(`static /+const+/ WNDPROC DateTimeProc;`));
-    mixin(gshared!(`static const TCHAR[] DateTimeClass = OS.DATETIMEPICK_CLASS;`));
-    mixin(gshared!(`static /+const+/ WNDPROC CalendarProc;`));
-    mixin(gshared!(`static const TCHAR[] CalendarClass = OS.MONTHCAL_CLASS;`));
+    __gshared static /+const+/ WNDPROC DateTimeProc;
+    __gshared static const TCHAR[] DateTimeClass = OS.DATETIMEPICK_CLASS;
+    __gshared static /+const+/ WNDPROC CalendarProc;
+    __gshared static const TCHAR[] CalendarClass = OS.MONTHCAL_CLASS;
 
-    mixin(gshared!(`private static bool static_this_completed = false;`));
-    mixin(sharedStatic_This!(`{
+    __gshared private static bool static_this_completed = false;
+    shared private static void static_this(){
         if( static_this_completed ){
             return;
         }
@@ -144,7 +144,7 @@
             OS.HeapFree (hHeap, 0, lpszClassName);
             static_this_completed = true;
         }
-    }`));
+    }
 
     static const int MARGIN = 4;
     static const int MAX_DIGIT = 9;
diff -r a72d9ca94ce2 org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/Display.d
--- a/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/Display.d	Sat Dec 24 17:16:18 2011 +0900
+++ b/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/Display.d	Sat Dec 24 18:02:52 2011 +0900
@@ -148,7 +148,7 @@
     //int windowProc_;
     int threadId;
     StringT windowClass_, windowShadowClass;
-    mixin(gshared!(`static int WindowClassCount;`));
+    __gshared static int WindowClassCount;
     static const String WindowName = "SWT_Window"; //$NON-NLS-1$
     static const String WindowShadowName = "SWT_WindowShadow"; //$NON-NLS-1$
     EventTable eventTable, filterTable;
@@ -161,11 +161,11 @@
     HWND lastGetHwnd;
     Control [] controlTable;
     static const int GROW_SIZE = 1024;
-    mixin(gshared!(`private static /+const+/ int SWT_OBJECT_INDEX;`));
-    mixin(gshared!(`static const bool USE_PROPERTY = !OS.IsWinCE;`));
-
-    mixin(gshared!(`private static bool static_this_completed = false;`));
-    mixin(sharedStatic_This!(`{
+    __gshared private static /+const+/ int SWT_OBJECT_INDEX;
+    __gshared static const bool USE_PROPERTY = !OS.IsWinCE;
+
+    __gshared private static bool static_this_completed = false;
+    shared private static void static_this(){
         if( static_this_completed ){
             return;
         }
@@ -182,10 +182,10 @@
             static_this_DeviceFinder ();
             static_this_completed = true;
         }
-    }`));
+    }
 
     /* Startup info */
-    mixin(gshared!(`private static STARTUPINFO* lpStartupInfo;`));
+    __gshared private static STARTUPINFO* lpStartupInfo;
     private static void static_this_StartupInfo (){
         static if (!OS.IsWinCE) {
             lpStartupInfo = new STARTUPINFO ();
@@ -321,7 +321,7 @@
     Object [] values;
 
     /* Key Mappings */
-    mixin(gshared!(`static const int [] [] KeyTable = [
+    __gshared static const int [] [] KeyTable = [
 
         /* Keyboard and Mouse Masks */
         [OS.VK_MENU,    SWT.ALT],
@@ -398,11 +398,11 @@
         [OS.VK_SNAPSHOT,    SWT.PRINT_SCREEN],
 //      [OS.VK_????,        SWT.HELP],
 
-    ];`));
+    ];
 
     /* Multiple Displays */
-    mixin(gshared!(`static Display Default;`));
-    mixin(gshared!(`static Display [] Displays;`));
+    __gshared static Display Default;
+    __gshared static Display [] Displays;
 
     /* Multiple Monitors */
     org.eclipse.swt.widgets.Monitor.Monitor[] monitors = null;
@@ -414,16 +414,16 @@
     static bool TrimEnabled = false;
 
     /* Private SWT Window Messages */
-    mixin(gshared!(`static const int SWT_GETACCELCOUNT  = OS.WM_APP;`));
-    mixin(gshared!(`static const int SWT_GETACCEL       = OS.WM_APP + 1;`));
-    mixin(gshared!(`static const int SWT_KEYMSG         = OS.WM_APP + 2;`));
-    mixin(gshared!(`static const int SWT_DESTROY        = OS.WM_APP + 3;`));
-    mixin(gshared!(`static const int SWT_TRAYICONMSG    = OS.WM_APP + 4;`));
-    mixin(gshared!(`static const int SWT_NULL           = OS.WM_APP + 5;`));
-    mixin(gshared!(`static const int SWT_RUNASYNC       = OS.WM_APP + 6;`));
-    mixin(gshared!(`static int SWT_TASKBARCREATED;`));
-    mixin(gshared!(`static int SWT_RESTORECARET;`));
-    mixin(gshared!(`static int DI_GETDRAGIMAGE;`));
+    __gshared static const int SWT_GETACCELCOUNT  = OS.WM_APP;
+    __gshared static const int SWT_GETACCEL       = OS.WM_APP + 1;
+    __gshared static const int SWT_KEYMSG         = OS.WM_APP + 2;
+    __gshared static const int SWT_DESTROY        = OS.WM_APP + 3;
+    __gshared static const int SWT_TRAYICONMSG    = OS.WM_APP + 4;
+    __gshared static const int SWT_NULL           = OS.WM_APP + 5;
+    __gshared static const int SWT_RUNASYNC       = OS.WM_APP + 6;
+    __gshared static int SWT_TASKBARCREATED;
+    __gshared static int SWT_RESTORECARET;
+    __gshared static int DI_GETDRAGIMAGE;
 
     /* Workaround for Adobe Reader 7.0 */
     int hitCount;
diff -r a72d9ca94ce2 org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/FileDialog.d
--- a/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/FileDialog.d	Sat Dec 24 17:16:18 2011 +0900
+++ b/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/FileDialog.d	Sat Dec 24 18:02:52 2011 +0900
@@ -51,8 +51,8 @@
     bool overwrite = false;
     static const String FILTER = "*.*";
     static int BUFFER_SIZE = 1024 * 32;
-    mixin(gshared!(`static bool USE_HOOK = true;`));
-    mixin(sharedStaticThis!(`{
+    __gshared static bool USE_HOOK = true;
+    shared static this(){
         /*
         *  Feature in Vista.  When OFN_ENABLEHOOK is set in the
         *  save or open file dialog,  Vista uses the old XP look
@@ -64,7 +64,7 @@
         if (!OS.IsWinCE && OS.WIN32_VERSION >= OS.VERSION (6, 0)) {
             USE_HOOK = false;
         }
-    }`));
+    }
 
 /**
  * Constructs a new instance of this class given only its parent.
diff -r a72d9ca94ce2 org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/Group.d
--- a/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/Group.d	Sat Dec 24 17:16:18 2011 +0900
+++ b/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/Group.d	Sat Dec 24 18:02:52 2011 +0900
@@ -56,7 +56,7 @@
 
     String text = "";
     static const int CLIENT_INSET = 3;
-    mixin(gshared!(`private static /+const+/ WNDPROC GroupProc;`));
+    __gshared private static /+const+/ WNDPROC GroupProc;
     static if( OS.IsWinCE ){
         static const TCHAR[] GroupClass = "BUTTON\0";
     }
@@ -64,8 +64,8 @@
         static const TCHAR[] GroupClass = "SWT_GROUP\0";
     }
 
-    mixin(gshared!(`private static bool static_this_completed = false;`));
-    mixin(sharedStatic_This!(`{
+    __gshared private static bool static_this_completed = false;
+    shared private static void static_this(){
         if( static_this_completed ){
             return;
         }
@@ -109,7 +109,7 @@
             }
             static_this_completed = true;
         }
-    }`));
+    }
 
 /**
  * Constructs a new instance of this class given its parent
diff -r a72d9ca94ce2 org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/IME.d
--- a/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/IME.d	Sat Dec 24 17:16:18 2011 +0900
+++ b/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/IME.d	Sat Dec 24 18:02:52 2011 +0900
@@ -55,20 +55,20 @@
     int [] ranges;
     TextStyle [] styles;
 
-    mixin(gshared!(`static const int WM_MSIME_MOUSE;`));
+    __gshared static const int WM_MSIME_MOUSE;
 
-    mixin(gshared!(`static byte [16] IID_ITfInputProcessorProfiles;`));
-    mixin(gshared!(`static byte [16] IID_ITfDisplayAttributeProvider;`));
-    mixin(gshared!(`static byte [16] CLSID_TF_InputProcessorProfiles;`));
-    mixin(gshared!(`static byte [16] GUID_TFCAT_TIP_KEYBOARD;`));
-    mixin(sharedStaticThis!(`{
+    __gshared static byte [16] IID_ITfInputProcessorProfiles;
+    __gshared static byte [16] IID_ITfDisplayAttributeProvider;
+    __gshared static byte [16] CLSID_TF_InputProcessorProfiles;
+    __gshared static byte [16] GUID_TFCAT_TIP_KEYBOARD;
+    shared static this(){
         WM_MSIME_MOUSE = OS.RegisterWindowMessage (StrToTCHARz ("MSIMEMouseOperation"));
 
         OS.IIDFromString ("{1F02B6C5-7842-4EE6-8A0B-9A24183A95CA}\0"w.ptr, IID_ITfInputProcessorProfiles.ptr);
         OS.IIDFromString ("{fee47777-163c-4769-996a-6e9c50ad8f54}\0"w.ptr, IID_ITfDisplayAttributeProvider.ptr);
         OS.IIDFromString ("{33C53A50-F456-4884-B049-85FD643ECFED}\0"w.ptr, CLSID_TF_InputProcessorProfiles.ptr);
         OS.IIDFromString ("{34745C63-B2F0-4784-8B67-5E12C8701A31}\0"w.ptr, GUID_TFCAT_TIP_KEYBOARD.ptr);
-    }`));
+    }
 
     /* TextLayout has a copy of these constants */
     static const int UNDERLINE_IME_DOT = 1 << 16;
diff -r a72d9ca94ce2 org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/Label.d
--- a/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/Label.d	Sat Dec 24 17:16:18 2011 +0900
+++ b/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/Label.d	Sat Dec 24 18:02:52 2011 +0900
@@ -70,11 +70,11 @@
     Image image;
     static const int MARGIN = 4;
     static const bool IMAGE_AND_TEXT = false;
-    mixin(gshared!(`private static /+const+/ WNDPROC LabelProc;`));
+    __gshared private static /+const+/ WNDPROC LabelProc;
     static const TCHAR[] LabelClass = "STATIC\0";
 
-    mixin(gshared!(`private static bool static_this_completed = false;`));
-    mixin(sharedStatic_This!(`{
+    __gshared private static bool static_this_completed = false;
+    shared private static void static_this(){
         if( static_this_completed ){
             return;
         }
@@ -87,7 +87,7 @@
             LabelProc = lpWndClass.lpfnWndProc;
             static_this_completed = true;
         }
-    }`));
+    }
 
 
 /**
diff -r a72d9ca94ce2 org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/Link.d
--- a/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/Link.d	Sat Dec 24 17:16:18 2011 +0900
+++ b/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/Link.d	Sat Dec 24 18:02:52 2011 +0900
@@ -75,12 +75,12 @@
     int [] mnemonics;
     int focusIndex, mouseDownIndex;
     HFONT font;
-    mixin(gshared!(`static /+const+/ RGB LINK_FOREGROUND;`));
-    mixin(gshared!(`static /+const+/ WNDPROC LinkProc;`));
-    mixin(gshared!(`static const TCHAR[] LinkClass = OS.WC_LINK;`));
+    __gshared static /+const+/ RGB LINK_FOREGROUND;
+    __gshared static /+const+/ WNDPROC LinkProc;
+    __gshared static const TCHAR[] LinkClass = OS.WC_LINK;
 
-    mixin(gshared!(`private static bool static_this_completed = false;`));
-    mixin(sharedStatic_This!(`{
+    __gshared private static bool static_this_completed = false;
+    shared private static void static_this(){
         if( static_this_completed ){
             return;
         }
@@ -126,7 +126,7 @@
             }
             static_this_completed = true;
         }
-    }`));
+    }
 
 /**
  * Constructs a new instance of this class given its parent
diff -r a72d9ca94ce2 org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/List.d
--- a/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/List.d	Sat Dec 24 17:16:18 2011 +0900
+++ b/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/List.d	Sat Dec 24 18:02:52 2011 +0900
@@ -55,11 +55,11 @@
     alias Scrollable.windowProc windowProc;
 
     static const int INSET = 3;
-    mixin(gshared!(`private static /+const+/ WNDPROC ListProc;`));
+    __gshared private static /+const+/ WNDPROC ListProc;
     static const TCHAR[] ListClass = "LISTBOX";
 
     private static bool static_this_completed = false;
-    mixin(sharedStatic_This!(`{
+    shared private static void static_this(){
         if( static_this_completed ){
             return;
         }
@@ -72,7 +72,7 @@
             ListProc = lpWndClass.lpfnWndProc;
             static_this_completed = true;
         }
-    }`));
+    }
 
 /**
  * Constructs a new instance of this class given its parent
diff -r a72d9ca94ce2 org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/MenuItem.d
--- a/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/MenuItem.d	Sat Dec 24 17:16:18 2011 +0900
+++ b/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/MenuItem.d	Sat Dec 24 18:02:52 2011 +0900
@@ -65,7 +65,7 @@
     * accelerator text.  The fix is to use smaller margins
     * everywhere but on Windows 98.
     */
-    mixin(gshared!(`private static int MARGIN_WIDTH_;`));
+    __gshared private static int MARGIN_WIDTH_;
     public static int MARGIN_WIDTH(){
         assert( static_this_completed );
         return MARGIN_WIDTH_;
@@ -76,8 +76,8 @@
         return MARGIN_HEIGHT_;
     }
 
-    mixin(gshared!(`private static bool static_this_completed = false;`));
-    mixin(sharedStatic_This!(`{
+    __gshared private static bool static_this_completed = false;
+    shared private static void static_this(){
         if( static_this_completed ){
             return;
         }
@@ -89,7 +89,7 @@
             MARGIN_HEIGHT_ = OS.IsWin95 ? 2 : 1;
             static_this_completed = true;
         }
-    }`));
+    }
 
 /**
  * Constructs a new instance of this class given its parent
diff -r a72d9ca94ce2 org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/ProgressBar.d
--- a/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/ProgressBar.d	Sat Dec 24 17:16:18 2011 +0900
+++ b/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/ProgressBar.d	Sat Dec 24 18:02:52 2011 +0900
@@ -52,11 +52,11 @@
     static const int DELAY = 100;
     static const int TIMER_ID = 100;
     static const int MINIMUM_WIDTH = 100;
-    mixin(gshared!(`private static /+const+/ WNDPROC ProgressBarProc;`));
+    __gshared private static /+const+/ WNDPROC ProgressBarProc;
     static const TCHAR[] ProgressBarClass = OS.PROGRESS_CLASS;
 
-    mixin(gshared!(`private static bool static_this_completed = false;`));
-    mixin(sharedStatic_This!(`{
+    __gshared private static bool static_this_completed = false;
+    shared private static void static_this(){
         if( static_this_completed ){
             return;
         }
@@ -97,7 +97,7 @@
             OS.HeapFree (hHeap, 0, lpszClassName);
             static_this_completed = true;
         }
-    }`));
+    }
 
 /**
  * Constructs a new instance of this class given its parent
diff -r a72d9ca94ce2 org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/Scale.d
--- a/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/Scale.d	Sat Dec 24 17:16:18 2011 +0900
+++ b/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/Scale.d	Sat Dec 24 18:02:52 2011 +0900
@@ -55,11 +55,11 @@
     alias Control.windowProc windowProc;
 
     bool ignoreResize, ignoreSelection;
-    mixin(gshared!(`private static /+const+/ WNDPROC TrackBarProc;`));
-    mixin(gshared!(`static const TCHAR[] TrackBarClass = OS.TRACKBAR_CLASS;`));
+    __gshared private static /+const+/ WNDPROC TrackBarProc;
+    __gshared static const TCHAR[] TrackBarClass = OS.TRACKBAR_CLASS;
 
-    mixin(gshared!(`private static bool static_this_completed = false;`));
-    mixin(sharedStatic_This!(`{
+    __gshared private static bool static_this_completed = false;
+    shared private static void static_this(){
         if( static_this_completed ){
             return;
         }
@@ -100,7 +100,7 @@
             OS.HeapFree (hHeap, 0, lpszClassName);
             static_this_completed = true;
         }
-    }`));
+    }
 
 /**
  * Constructs a new instance of this class given its parent
diff -r a72d9ca94ce2 org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/Shell.d
--- a/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/Shell.d	Sat Dec 24 17:16:18 2011 +0900
+++ b/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/Shell.d	Sat Dec 24 18:02:52 2011 +0900
@@ -156,8 +156,8 @@
     static if( OS.IsWinCE ){
         SHACTIVATEINFO psai;
     }
-    mixin(gshared!(`static /+const+/ WNDPROC ToolTipProc;`));
-    mixin(gshared!(`static /+const+/ WNDPROC DialogProc;`));
+    __gshared static /+const+/ WNDPROC ToolTipProc;
+    __gshared static /+const+/ WNDPROC DialogProc;
     static if( OS.IsWinCE ){
         static const StringT DialogClass = "Dialog\0"w;
     }
@@ -178,8 +178,8 @@
     ];
     const static int BRUSHES_SIZE = 32;
 
-    mixin(gshared!(`private static bool static_this_completed = false;`));
-    mixin(sharedStatic_This!(`{
+    __gshared private static bool static_this_completed = false;
+    shared private static void static_this(){
         if( static_this_completed ){
             return;
         }
@@ -192,7 +192,7 @@
             DialogProc = lpWndClass.lpfnWndProc;
             static_this_completed = true;
         }
-    }`));
+    }
 
 /**
  * Constructs a new instance of this class. This is equivalent
diff -r a72d9ca94ce2 org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/Slider.d
--- a/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/Slider.d	Sat Dec 24 17:16:18 2011 +0900
+++ b/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/Slider.d	Sat Dec 24 18:02:52 2011 +0900
@@ -85,11 +85,11 @@
 
     int increment, pageIncrement;
     bool ignoreFocus;
-    mixin(gshared!(`static /+const+/ WNDPROC ScrollBarProc;`));
+    __gshared static /+const+/ WNDPROC ScrollBarProc;
     static const TCHAR[] ScrollBarClass = "SCROLLBAR";
 
-    mixin(gshared!(`private static bool static_this_completed = false;`));
-    mixin(sharedStatic_This!(`{
+    __gshared private static bool static_this_completed = false;
+    shared private static void static_this(){
         if( static_this_completed ){
             return;
         }
@@ -102,7 +102,7 @@
             ScrollBarProc = lpWndClass.lpfnWndProc;
             static_this_completed = true;
         }
-    }`));
+    }
 
 /**
  * Constructs a new instance of this class given its parent
diff -r a72d9ca94ce2 org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/Spinner.d
--- a/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/Spinner.d	Sat Dec 24 17:16:18 2011 +0900
+++ b/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/Spinner.d	Sat Dec 24 18:02:52 2011 +0900
@@ -64,13 +64,13 @@
     HWND hwndText, hwndUpDown;
     bool ignoreModify;
     int pageIncrement, digits;
-    mixin(gshared!(`static /+const+/ WNDPROC EditProc;`));
+    __gshared static /+const+/ WNDPROC EditProc;
     static const TCHAR[] EditClass = "EDIT";
-    mixin(gshared!(`static /+const+/ WNDPROC UpDownProc;`));
-    mixin(gshared!(`static const TCHAR[] UpDownClass = OS.UPDOWN_CLASS;`));
+    __gshared static /+const+/ WNDPROC UpDownProc;
+    __gshared static const TCHAR[] UpDownClass = OS.UPDOWN_CLASS;
 
-    mixin(gshared!(`private static bool static_this_completed = false;`));
-    mixin(sharedStatic_This!(`{
+    __gshared private static bool static_this_completed = false;
+    shared private static void static_this(){
         if( static_this_completed ){
             return;
         }
@@ -85,23 +85,23 @@
             UpDownProc = lpWndClass.lpfnWndProc;
             static_this_completed = true;
         }
-    }`));
+    }
     /**
      * the operating system limit for the number of characters
      * that the text field in an instance of this class can hold
      *
      * @since 3.4
      */
-     mixin(gshared!(`public static int LIMIT;`));
+     __gshared public static int LIMIT;
 
     /*
      * These values can be different on different platforms.
      * Therefore they are not initialized in the declaration
      * to stop the compiler from inlining.
      */
-    mixin(sharedStaticThis!(`{
+    shared static this(){
         LIMIT = OS.IsWinNT ? 0x7FFFFFFF : 0x7FFF;
-    }`));
+    }
 
 /**
  * Constructs a new instance of this class given its parent
diff -r a72d9ca94ce2 org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/TabFolder.d
--- a/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/TabFolder.d	Sat Dec 24 17:16:18 2011 +0900
+++ b/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/TabFolder.d	Sat Dec 24 18:02:52 2011 +0900
@@ -68,8 +68,8 @@
 
     TabItem [] items;
     ImageList imageList;
-    mixin(gshared!(`private static /+const+/ WNDPROC TabFolderProc;`));
-    mixin(gshared!(`static const TCHAR[] TabFolderClass = OS.WC_TABCONTROL;`));
+    __gshared private static /+const+/ WNDPROC TabFolderProc;
+    __gshared static const TCHAR[] TabFolderClass = OS.WC_TABCONTROL;
 
     /*
     * These are the undocumented control id's for the children of
@@ -78,8 +78,8 @@
     */
     static const int ID_UPDOWN = 1;
 
-    mixin(gshared!(`private static bool static_this_completed = false;`));
-    mixin(sharedStatic_This!(`{
+    __gshared private static bool static_this_completed = false;
+    shared private static void static_this(){
         if( static_this_completed ){
             return;
         }
@@ -121,7 +121,7 @@
             OS.HeapFree (hHeap, 0, lpszClassName);
             static_this_completed = true;
         }
-    }`));
+    }
 
 /**
  * Constructs a new instance of this class given its parent
diff -r a72d9ca94ce2 org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/Table.d
--- a/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/Table.d	Sat Dec 24 17:16:18 2011 +0900
+++ b/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/Table.d	Sat Dec 24 18:02:52 2011 +0900
@@ -108,7 +108,7 @@
     bool customDraw, dragStarted, explorerTheme, firstColumnImage, fixScrollWidth, tipRequested, wasSelected, wasResized;
     bool ignoreActivate, ignoreSelect, ignoreShrink, ignoreResize, ignoreColumnMove, ignoreColumnResize, fullRowSelect;
     int itemHeight, lastIndexOf, lastWidth, sortDirection, resizeCount, selectionForeground, hotIndex;
-    mixin(gshared!(`static /*final*/ WNDPROC HeaderProc;`));
+    __gshared static /*final*/ WNDPROC HeaderProc;
     static const int INSET = 4;
     static const int GRID_WIDTH = 1;
     static const int SORT_WIDTH = 10;
@@ -120,11 +120,11 @@
     static const int V_SCROLL_LIMIT = 16;
     static const int DRAG_IMAGE_SIZE = 301;
     static const bool EXPLORER_THEME = true;
-    mixin(gshared!(`private static /+const+/ WNDPROC TableProc;`));
-    mixin(gshared!(`static const TCHAR[] TableClass = OS.WC_LISTVIEW;`));
-
-    mixin(gshared!(`private static bool static_this_completed = false;`));
-    mixin(sharedStatic_This!(`{
+    __gshared private static /+const+/ WNDPROC TableProc;
+    __gshared static const TCHAR[] TableClass = OS.WC_LISTVIEW;
+
+    __gshared private static bool static_this_completed = false;
+    shared private static void static_this(){
         if( static_this_completed ){
             return;
         }
@@ -137,7 +137,7 @@
             TableProc = lpWndClass.lpfnWndProc;
             static_this_completed = true;
         }
-    }`));
+    }
 
 /**
  * Constructs a new instance of this class given its parent
diff -r a72d9ca94ce2 org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/Text.d
--- a/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/Text.d	Sat Dec 24 17:16:18 2011 +0900
+++ b/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/Text.d	Sat Dec 24 18:02:52 2011 +0900
@@ -80,7 +80,7 @@
     * the native widget implementation.
     * </p>
     */
-    mixin(gshared!(`private static int LIMIT_;`));
+    __gshared private static int LIMIT_;
     public static int LIMIT(){
         assert( static_this_completed );
         return LIMIT_;
@@ -104,11 +104,11 @@
     * to stop the compiler from inlining.
     */
 
-    mixin(gshared!(`private static /+const+/ WNDPROC EditProc;`));
+    __gshared private static /+const+/ WNDPROC EditProc;
     static const TCHAR[] EditClass = "EDIT\0";
 
-    mixin(gshared!(`private static bool static_this_completed = false;`));
-    mixin(sharedStatic_This!(`{
+    __gshared private static bool static_this_completed = false;
+    shared private static void static_this(){
         if( static_this_completed ){
             return;
         }
@@ -137,7 +137,7 @@
     //      PASSWORD = echo !is 0 ? echo : '*';
             static_this_completed = true;
         }
-    }`));
+    }
 
 /**
  * Constructs a new instance of this class given its parent
diff -r a72d9ca94ce2 org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/ToolBar.d
--- a/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/ToolBar.d	Sat Dec 24 17:16:18 2011 +0900
+++ b/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/ToolBar.d	Sat Dec 24 18:02:52 2011 +0900
@@ -67,11 +67,11 @@
     ToolItem [] items;
     bool ignoreResize, ignoreMouse;
     ImageList imageList, disabledImageList, hotImageList;
-    mixin(gshared!(`private static /+const+/ WNDPROC ToolBarProc;`));
-    mixin(gshared!(`static const TCHAR[] ToolBarClass = OS.TOOLBARCLASSNAME;`));
+    __gshared private static /+const+/ WNDPROC ToolBarProc;
+    __gshared static const TCHAR[] ToolBarClass = OS.TOOLBARCLASSNAME;
 
-    mixin(gshared!(`private static bool static_this_completed = false;`));
-    mixin(sharedStatic_This!(`{
+    __gshared private static bool static_this_completed = false;
+    shared private static void static_this(){
         if( static_this_completed ){
             return;
         }
@@ -84,7 +84,7 @@
             ToolBarProc = lpWndClass.lpfnWndProc;
             static_this_completed = true;
         }
-    }`));
+    }
 
 
 
diff -r a72d9ca94ce2 org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/Tree.d
--- a/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/Tree.d	Sat Dec 24 17:16:18 2011 +0900
+++ b/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/Tree.d	Sat Dec 24 18:02:52 2011 +0900
@@ -135,13 +135,13 @@
     static const int EXPLORER_EXTRA = 2;
     static const int DRAG_IMAGE_SIZE = 301;
     static const bool EXPLORER_THEME = true;
-    mixin(gshared!(`private static /+const+/ WNDPROC TreeProc;`));
-    mixin(gshared!(`static const TCHAR[] TreeClass = OS.WC_TREEVIEW;`));
-    mixin(gshared!(`private static /+const+/ WNDPROC HeaderProc;`));
-    mixin(gshared!(`static const TCHAR[] HeaderClass = OS.WC_HEADER;`));
-
-    mixin(gshared!(`private static bool static_this_completed = false;`));
-    mixin(sharedStatic_This!(`{
+    __gshared private static /+const+/ WNDPROC TreeProc;
+    __gshared static const TCHAR[] TreeClass = OS.WC_TREEVIEW;
+    __gshared private static /+const+/ WNDPROC HeaderProc;
+    __gshared static const TCHAR[] HeaderClass = OS.WC_HEADER;
+
+    __gshared private static bool static_this_completed = false;
+    shared private static void static_this(){
         if( static_this_completed ){
             return;
         }
@@ -156,9 +156,9 @@
             HeaderProc = lpWndClass.lpfnWndProc;
             static_this_completed = true;
         }
-    }`));
-
-    mixin(gshared!(`private static Tree sThis;`));
+    }
+
+    __gshared private static Tree sThis;
 /**
  * Constructs a new instance of this class given its parent
  * and a style value describing its behavior and appearance.
diff -r a72d9ca94ce2 org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/Widget.d
--- a/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/Widget.d	Sat Dec 24 17:16:18 2011 +0900
+++ b/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/Widget.d	Sat Dec 24 18:02:52 2011 +0900
@@ -112,8 +112,8 @@
     /* Check and initialize the Common Controls DLL */
     static const int MAJOR = 5, MINOR = 80;
 
-    mixin(gshared!(`private static bool static_this_completed = false;`));
-    mixin(sharedStatic_This!(`{
+    __gshared private static bool static_this_completed = false;
+    shared private static void static_this(){
         if( static_this_completed ){
             return;
         }
@@ -130,7 +130,7 @@
             OS.InitCommonControls ();
             static_this_completed = true;
         }
-    }`));
+    }
 
 /**
  * Prevents uninitialized instances from being created outside the package.
